### PR TITLE
remove redundant doc and add tutorial

### PR DESCRIPTION
**Pull request by qingyun-wu**
## Why are these changes needed?

Close https://github.com/microsoft/FLAML/issues/1003
Adding a tutorial folder. This folder is intended to include a summary page as an .md file for each upcoming and past tutorials on FLAML. The code examples and notebooks in each summary page may be contained in a specific branch.

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: tutorials/flaml-tutorial-pydata-23.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+In this session, we will give a hands-on tutorial on (1) how to use FLAML to automate typical machine learning tasks and generic tuning on user-defined functions; (2) how to make the best use of FLAML to perform AutoML and tuning in various applications with the help of rich customization choices and advanced functionalities provided by FLAML; and (3) several new features of FLAML based on our latest research and development work around FLAML.
+
+Target audience and prerequisites. The primary target audience of this tutorial is machine learning and data science practitioners, including data scientists, machine learning engineers, domain experts, and researchers, especially researchers who are interested in the automation of machine learning or artificial intelligence methods. The audiences are assumed to have the basic knowledge of machine learning, data science, and Python programming language.
+
+Strategies and needs of the hands-on tutorial. This tutorial is intended to be hands-on and interactive. Most of the topics covered in the tutorial will be accompanied by demonstrations implemented in Jupyter notebooks.
+
+### Tutorial Outline
```
Comment (thinkall) [inline tutorials/flaml-tutorial-pydata-23.md:30]: The tutorial outline is updated, what about we wait two more days and update this part according to the final presentation?

File: tutorials/flaml-tutorial-aaai-23.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## Session Agenda
+
+### **Part 1. Overview of FLAML**
+
+- Overview of AutoML and FLAML
+- Basic usages of FLAML
+    - Task-oriented AutoML
+        - [Documentation](https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML)
+        - [Notebook: A classification task with AutoML](https://github.com/microsoft/FLAML/blob/tutorial-aaai23/notebook/automl_classification.ipynb); [Open In Colab](https://colab.research.google.com/github/microsoft/FLAML/blob/tutorial-aaai23/notebook/automl_classification.ipynb)
```
Comment (thinkall) [inline tutorials/flaml-tutorial-aaai-23.md:34]: Maybe we host the notebooks for each tutorial in a separate subfolder under the `tutorials` folder and update the links. Thus we can have everything in the `main` branch.

**Comment by qingyun-wu** [comment]
@thinkall. I updated the links to the notebook in pydata tutorial's readme file. Could you help correct the invalid links in the tutorial slides accordingly? 

Here are the places in the slides I identified that need changes: 

1. Check and update the links on page 32 and page 40:
I cannot open the links. 

2. The links on page 48 are dated and cannot be found anymore because of recent changes in FLAML. 
Could you update the links on page 48 to the following ones:

Documentation:
https://microsoft.github.io/FLAML/docs/Use-Cases/Auto-Generation

Demo Notebook - Code
https://github.com/microsoft/FLAML/blob/main/notebook/autogen_openai.ipynb

Deme Notebook - Math
https://github.com/microsoft/FLAML/blob/main/notebook/autogen_chatgpt.ipynb

**Comment by thinkall** [comment]
> Thank you @thinkall ! I updated the links to the notebook in pydata tutorial's readme file. Could you help correct the invalid links in the tutorial slides accordingly?
> 
> Here are the places in the slides I identified that need changes:
> 
> 1. Check and update the links on page 32 and page 40:
>    I cannot open the links.
> 2. The links on page 48 are dated and cannot be found anymore because of recent changes in FLAML.
>    Could you update the links on page 48 to the following ones:
> 
> Documentation: https://microsoft.github.io/FLAML/docs/Use-Cases/Auto-Generation
> 
> Demo Notebook - Code https://github.com/microsoft/FLAML/blob/main/notebook/autogen_openai.ipynb
> 
> Deme Notebook - Math https://github.com/microsoft/FLAML/blob/main/notebook/autogen_chatgpt.ipynb

Thank you @qingyun-wu . The links in the slides have been updated.

File: tutorials/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Please find upcoming and past tutorials on FLAML below:
+- [PyData Seattle 2023](flaml-tutorial-pydata-23.md)
+- [A hands-on tutorial on FLAML presented at KDD 2022](flaml-tutorial-kdd-22.md)
```
Comment (sonichi) [inline tutorials/README.md:3]: add PyData'23

File: tutorials/pydata2023seattle/demo_1_flight_delays_automl.ipynb
Comment (sonichi) [inline tutorials/pydata2023seattle/demo_1_flight_delays_automl.ipynb:1]: move to notebooks/ ?
and use a more consistent name w/ other notebooks?

File: tutorials/README.md (hunk 3 above)
Comment (thinkall) [reply tutorials/README.md:3]: Sorry, I'm not sure I understand. PyData'23 is already here.

File: tutorials/pydata2023seattle/demo_1_flight_delays_automl.ipynb
Comment (thinkall) [reply tutorials/pydata2023seattle/demo_1_flight_delays_automl.ipynb:1]: Moved. As I checked the links in aaai23 and kdd22, notebooks are linked to different branches. Maybe we should move them all to the main branch too?

Btw, for different tutorial sessions, I think we'll adpot the tutorial notebooks a little bit. Therefore, we'll have multi-notebooks with only a few different cells. Such as the `automl_classification.ipynb` and `automl_flight_delays.ipynb`. Maybe it's better to keep the notebooks  as-is for different sessions, and gather them in subfolders of `tutorials`, while in `notebook` folder, we keep the base notebooks.

**Review by sonichi — COMMENTED** [review]
We can use commit version in the link to pin down the different versions, without needing to create separate folders or branches per tutorial. No need to modify the old tutorials though.

File: notebook/automl_bankrupt_synapseml.ipynb
[the remark this answers is not in the document]
Comment (qingyun-wu) [reply notebook/automl_bankrupt_synapseml.ipynb:1]: Updated. thanks!

File: tutorials/flaml-tutorial-pydata-23.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+- Advanced Functionalities
+- Parallelization with Apache Spark
+    - [Notebook: FLAML AutoML on Apache Spark](https://github.com/microsoft/FLAML/blob/d047c79352a2b5d32b72f4323dadfa2be0db8a45/notebook/automl_bankrupt_synapseml.ipynb)
+
+### **Part 4. New features in FLAML**
+- Targeted Hyperparameter Optimization With Lexicographic Objectives
+    - [Notebook: Tune models with lexicographic preference across objectives](https://github.com/microsoft/FLAML/blob/7ae410c8eb967e2084b2e7dbe7d5fa2145a44b79/notebook/tune_lexicographic.ipynb)
+- OpenAI GPT-3, GPT-4 and ChatGPT tuning
+    - [Notebook: Use FLAML to Tune OpenAI Models](https://github.com/microsoft/FLAML/blob/7ae410c8eb967e2084b2e7dbe7d5fa2145a44b79/notebook/autogen_openai_completion.ipynb)
+    - [Notebook: Use FLAML to Tune ChatGPT](https://github.com/microsoft/FLAML/blob/7ae410c8eb967e2084b2e7dbe7d5fa2145a44b79/notebook/autogen_chatgpt_gpt4.ipynb)
```
Comment (sonichi) [inline tutorials/flaml-tutorial-pydata-23.md:40]: Could you update these two links with the latest version?

Comment (qingyun-wu) [reply tutorials/flaml-tutorial-pydata-23.md]: Updated. Thanks!

**Review by thinkall — APPROVED** [review]

File: tutorials/flaml-tutorial-pydata-23.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- Advanced Functionalities
+- Parallelization with Apache Spark
+    - [Notebook: FLAML AutoML on Apache Spark](https://github.com/microsoft/FLAML/blob/d047c79352a2b5d32b72f4323dadfa2be0db8a45/notebook/automl_bankrupt_synapseml.ipynb)
+
+### **Part 4. New features in FLAML**
+- Targeted Hyperparameter Optimization With Lexicographic Objectives
+    - [Notebook: Tune models with lexicographic preference across objectives](https://github.com/microsoft/FLAML/blob/7ae410c8eb967e2084b2e7dbe7d5fa2145a44b79/notebook/tune_lexicographic.ipynb)
+- OpenAI GPT-3, GPT-4 and ChatGPT tuning
+    - [Notebook: Use FLAML to Tune OpenAI Models](https://github.com/microsoft/FLAML/blob/main/notebook/autogen_openai_completion.ipynb)
+    - [Notebook: Use FLAML to Tune ChatGPT](https://github.com/microsoft/FLAML/blob/main/notebook/autogen_chatgpt_gpt4.ipynb)
```
Comment (sonichi) [inline tutorials/flaml-tutorial-pydata-23.md:40]: I mean, use the latest pinned version.

**Review by sonichi — APPROVED** [review]

**Pull request merged**